src/OrbitGl/CaptureWindowTest.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "CaptureWindow.h"
+#include "UnitTestSlider.h"
+
+namespace orbit_gl {
+
+class UnitTestCaptureWindow : public CaptureWindow, public testing::Test {
+ public:
+  explicit UnitTestCaptureWindow() : CaptureWindow(nullptr) {
+    slider_ = std::make_unique<UnitTestSlider>(viewport_, false);
+    vertical_slider_ = std::make_unique<UnitTestSlider>(viewport_, true);
+  }
+};
+
+TEST_F(UnitTestCaptureWindow, SlidersRespondToMouseOver) {
+  Resize(100, 200);
+
+  GlSlider* slider = FindSliderUnderMouseCursor(50, 50);
+  EXPECT_EQ(nullptr, slider);
+  slider = FindSliderUnderMouseCursor(95, 10);
+  EXPECT_EQ(vertical_slider_.get(), slider);
+  slider = FindSliderUnderMouseCursor(5, 195);
+  EXPECT_EQ(slider_.get(), slider);
+  slider = FindSliderUnderMouseCursor(95, 195);
+  EXPECT_EQ(nullptr, slider);
+
+  MouseMoved(95, 10, false, false, false);
+  EXPECT_TRUE(dynamic_cast<UnitTestSlider*>(vertical_slider_.get())->IsMouseOver());
```

Comment:
Could you store the results of the dynamic_casts into variables and use them in all the EXPECT statements? Maybe also add an ASSERT which checks that the dynamic_cast doesn't return a nullptr?

Or what Pierric said. :-)

Reply:
Yep, I've changed that to store them in a variable and casting only once.